src/main/java/cow/farm/utils/LukaszGenericArrayList.java

```diff
@@ -98,12 +96,7 @@ public boolean containsAll(Collection c) {
 
 	@Override
 	public int indexOf(Object element) {
-		for (int i = 0; i < array.length; i++) {
-			// copy paste
-			if (element == null ? get(i) == null : element.equals(get(i)))
-				return i;
-		}
-		return -1;
+		return lastIndexOf(element);
```

Comment:
you should see javadoc for this method in interface, indexOf means something different than lastIndexOf. If you want to remove copy paste you can also extract to some private method that has usage in both public methods.